src/Products/PluginIndexes/DateRangeIndex/DateRangeIndex.py

```diff
@@ -238,12 +228,24 @@ def uniqueValues(self, name=None, withLengths=0):
                     else:
                         yield (key, len(value))
 
-    def _cache_key(self, catalog):
-        cid = catalog.getId()
-        counter = getattr(aq_base(catalog), 'getCounter', None)
-        if counter is not None:
-            return '%s_%s' % (cid, counter())
-        return cid
+    def _record_cache_key(self, record, resultset=None):
+        term = self._convertDateTime(record.keys[0])
+
+        # why flatten to 10 minutes? 'term' is already flattened
+        # to minutes by _convertDateTime.
+        # tid = isinstance(term, int) and term / 10 or 'None'
```

Comment:
I'd just remove this bit. I think I explained in a comment elsewhere why this was added, but its an edge-case. It's trying to let the cache kick in, if a requests takes longer than a couple seconds and crosses over the minute boundary. If the "same" query is issued, but with a slightly different time constraint at 15:01:59 and again 15:02:01 the cache key is different, as it based on the minute value. But there is a chance that the result should actually be different, so this isn't a totally safe assumption.